models/user.go

```diff
@@ -2,22 +2,45 @@ package models
 
 import (
 	"fmt"
+
+	"github.com/wantedly/developers-account-mapper/services"
 )
 
-// User stores login user name and GitHub username
+// User stores login user name and accounts information
 type User struct {
-	LoginName string
+	LoginName      string
 	GitHubUsername string
+	SlackUsername  string
+	SlackUserId    string
 }
 
 // NewUser creates new User instance
-func NewUser(loginName string, githubUsername string) *User {
+func NewUser(loginName string, githubUsername string, slackUsername string, slackUserId string) *User {
 	return &User{
 		LoginName: loginName,
 		GitHubUsername: githubUsername,
+		SlackUsername: slackUsername,
+		SlackUserId: slackUserId,
 	}
 }
 
+func (u *User) SetSlackUserId(newId string) *User {
+	u.SlackUserId = newId
+	return u
+}
+
+func (u *User) RetrieveSlackUserId() (*User, error) {
+	nameIdMap, err := services.SlackUserList()
+	if err != nil {
+		return nil, err
+	}
+	u.SetSlackUserId(nameIdMap[u.SlackUsername])
+	return u, nil
+}
+
 func (u *User) String() string {
-	return fmt.Sprintf("%v:@%v", u.LoginName, u.GitHubUsername)
+	if u.SlackUserId == "" {
+		u.RetrieveSlackUserId()
```

Comment:
Rewrite to catch the error, but actually, I don't know how to handle this. Is that ok to return the error string?